pkg/runtime/interface.go

```diff
@@ -37,6 +37,8 @@ type Runtime interface {
 	// https://github.com/kubernetes/kubernetes/pull/129313 becomes available
 
 	NewObjects(ctx context.Context, trainJob *trainer.TrainJob) ([]any, error)
+	BuildRuntimeInfo(trainJob *trainer.TrainJob, jobSetTemplateSpec trainer.JobSetTemplateSpec, mlPolicy *trainer.MLPolicy, podGroupPolicy *trainer.PodGroupPolicy,
```

Comment:
```suggestion
	BuildRuntimeInfo(trainJob *trainer.TrainJob, runtimeTemplateSpec, any mlPolicy *trainer.MLPolicy, podGroupPolicy *trainer.PodGroupPolicy,
```

We should avoid exposing underlying Job typed throughout interface since underlying Job is not only for JobSet.